### DISCUSSION TITLE
Creating form with multiple fields

### DISCUSSION THREAD

**Discussion by JazzHoolahawk** [microsoft-search/pnp-modern-search#3190] [Q&A]
Hi there,

I have been tasked with creating a complex form field similar to the advance form that you got out of the box in previous versions of SharePoint (i.e 2013) to query the search index in SPO.

The web parts for PnP Modern search only has a single field. Is there a way to create multiple related form fields in a single web part for example to use PnP modern search? What would be the best way of doing this to use the PnP modern search?

Any help would be appreciated :)

**Comment by kasperbolarsen** [comment]
Hi,
As the Results web part can only be connected to one Search Box you are out of luck. The only option I see, is one where you have a number of custom fields, one for each of the properties you will be catering for, and a "Search" button to activate the search. The code behind the button will pick up the values in the custom fields and create a URL containing the search query, see https://microsoft-search.github.io/pnp-modern-search/scenarios/use-query-string-in-url/

It is an ugly design and if you look at the current offering from Microsoft Search it only provides ONE search box, so perhaps the requirement should be reevaluated/scraped?
